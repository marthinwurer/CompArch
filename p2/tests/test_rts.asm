;
; tests RTS
;
; uses instructions MOV, BPT, HALT
;
; uses addressing modes ?
;
	.globl	start
;
a:	.word	loc1
b:	.word	loc2
c:	.word	loc3
;
. = . + 100
;
start:	mov	#1000,sp
	mov	a,r1
	mov	b,r2
	mov	c,r3
	bpt
;
	rts	r1
;
. = . + 100
loc1:	bpt
	rts	r2
;
. = . + 100
loc2:	bpt
	rts	r3
;
. = . + 100
loc3:	bpt
	halt
;
	.end	start
