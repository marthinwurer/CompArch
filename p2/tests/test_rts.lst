      1                                 ;
      2                                 ; tests RTS
      3                                 ;
      4                                 ; uses instructions MOV, BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes ?
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000232'                 a:       .word   loc1
     11 000002  000336'                 b:       .word   loc2
     12 000004  000442'                 c:       .word   loc3
     13                                 ;
     14 000106                          . = . + 100
     15                                 ;
     16 000106  012706  001000          start:   mov     #1000,sp
     17 000112  016701  177662                   mov     a,r1
     18 000116  016702  177660                   mov     b,r2
     19 000122  016703  177656                   mov     c,r3
     20 000126  000003                           bpt
     21                                 ;
     22 000130  000201                           rts     r1
     23                                 ;
     24 000232                          . = . + 100
     25 000232  000003                  loc1:    bpt
     26 000234  000202                           rts     r2
     27                                 ;
     28 000336                          . = . + 100
     29 000336  000003                  loc2:    bpt
     30 000340  000203                           rts     r3
     31                                 ;
     32 000442                          . = . + 100
     33 000442  000003                  loc3:    bpt
     34 000444  000000                           halt
     35                                 ;
     36 000446  000106'                          .end    start


a        000000 R  
b        000002 R  
c        000004 R  
loc1     000232 R  
loc2     000336 R  
loc3     000442 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000106 RG 
