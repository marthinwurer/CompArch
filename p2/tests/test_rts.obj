000 2 9a 00
002 2 de 00
004 2 22 01
046 4 c6 15 00 02
04a 4 c1 1d b2 ff
04e 4 c2 1d b0 ff
052 4 c3 1d ae ff
056 2 03 00
058 2 81 00
09a 2 03 00
09c 2 82 00
0de 2 03 00
0e0 2 83 00
122 2 03 00
124 2 00 00
046
