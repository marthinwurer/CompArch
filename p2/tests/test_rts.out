CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 106
000106:  PS[00]  012706  MOV (R7)+,R6
000112:  PS[00]  016701 177662  MOV 177662(R7),R1
000116:  PS[00]  016702 177660  MOV 177660(R7),R2
000122:  PS[00]  016703 177656  MOV 177656(R7),R3
000126:  PS[00]  000003  BPT
  R0[000000]  R1[000232]  R2[000336]  R3[000442]
  R4[000000]  R5[000000]  R6[001000]  R7[000130]
000130:  PS[00]  000201  RTS R1
000232:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000336]  R3[000442]
  R4[000000]  R5[000000]  R6[001002]  R7[000234]
000234:  PS[00]  000202  RTS R2
000336:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000442]
  R4[000000]  R5[000000]  R6[001004]  R7[000340]
000340:  PS[00]  000203  RTS R3
000442:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[001006]  R7[000444]
000444:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 71 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
